airflow/providers/microsoft/azure/hooks/base_azure.py

```diff
@@ -0,0 +1,68 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from os import getenv
+from typing import Any
+
+from azure.common.client_factory import get_client_from_auth_file
+from azure.common.credentials import ServicePrincipalCredentials
+
+from airflow import AirflowException
+from airflow.hooks.base_hook import BaseHook
+
+
+class AzureBaseHook(BaseHook):
+    """
+    This hook acts as a base hook for azure services. It offers several authentication mechanisms to
+    authenticate the client library used for upstream azure hooks.
+
+    :param sdk_client: The SDKClient to use.
+    :param conn_id: The azure connection id which refers to the information to connect to the service.
+    """
+
+    def __init__(self, sdk_client: Any, conn_id: str = 'azure_default'):
+        self.sdk_client = sdk_client
+        self.conn_id = conn_id
+        super().__init__()
+
+    def get_conn(self) -> Any:
+        """
+        Authenticates the resource using the connection id passed during init.
+
+        :return: the authenticated client.
+        """
+        conn = self.get_connection(self.conn_id)
+
+        key_path = conn.extra_dejson.get('key_path') or getenv('AZURE_AUTH_LOCATION')
+        if key_path:
+            if not key_path.endswith('.json'):
+                raise AirflowException('Unrecognised extension for key file.')
+            self.log.info('Getting connection using a JSON key file.')
+            return get_client_from_auth_file(
+                client_class=self.sdk_client,
+                auth_path=key_path
+            )
+
+        self.log.info('Getting connection using specific credentials and subscription_id.')
+        return self.sdk_client(
+            credentials=ServicePrincipalCredentials(
+                client_id=conn.login,
+                secret=conn.password,
+                tenant=conn.extra_dejson['tenantId']
```

Comment:
Should we validate that `tenantId` and `subscriptionId` are set or else this will result in a KeyError

Reply:
1.)
Or just do this:
```suggestion
                tenant=conn.extra_dejson.get('tenantId')
``` 
so azure raises the error which will probably be more expressive?

2.)
Simply validate and raise Exception that the field is missing.

Reply:
I'm in favor of 1) this reduce the number of logic on our side

Reply:
Yeah (1) sounds better